development/lib/build-type.js

```diff
@@ -18,143 +17,252 @@ const {
   nullable,
   never,
   literal,
-} = require('superstruct');
+} = require('@metamask/superstruct');
 const yaml = require('yaml');
-const { uniqWith } = require('lodash');
+const { cloneDeep, merge, uniqWith } = require('lodash');
 
-const BUILDS_YML_PATH = path.resolve('./builds.yml');
+const BUILDS_YML_PATH = path.resolve(__dirname, '../../builds.yml');
 
+// This is in fact a valid jsdoc tag.
+/* eslint-disable jsdoc/check-tag-names */
 /**
- * @type {import('superstruct').Infer<typeof BuildTypesStruct> | null}
+ * @import { Infer, Struct, StructError } from '@metamask/superstruct';
  */
-let cachedBuildTypes = null;
+/* eslint-enable jsdoc/check-tag-names */
 
 /**
  * Ensures that the array item contains only elements that are distinct from each other
  *
- * @template {Struct<any>} Element
- * @type {import('./build-type').Unique<Element>}
+ * @template {Struct<unknown>} Element
+ * @typedef {(
+ *   struct: Struct<Infer<Element>[], Infer<Element>>,
+ *   eq?: (a: Infer<Element>, b: Infer<Element>) => boolean
+ * ) => Struct<Infer<Element>[], Infer<Element>>} Unique
+ */
+
+/**
+ * @type {Infer<typeof BuildTypesStruct> | null}
+ */
+let _cachedBuildTypes = null;
+
+/**
+ * Given a source array and a set or array of its unique values, returns the elements
+ * that are duplicated in the source array.
+ *
+ * @template {unknown} Element
+ * @param {Element[]} source
+ * @param {Set<Element> | Element[]} uniqueValues
+ * @returns {Element[]}
+ */
+const getDuplicates = (source, uniqueValues) => {
+  const uniqueValuesCopy = new Set(uniqueValues);
+  return source.filter((item) => {
+    if (uniqueValuesCopy.has(item)) {
+      uniqueValuesCopy.delete(item);
+      return false;
+    }
+    return true;
+  });
+};
```

Comment:
The function name implies duplicates will be returned yet the function returns a boolean.  Additionally, `returns the elements that are duplicated in the source array.` also implies duplicates will be returned.

Reply:
Ah, the function works as described. The booleans are returned in the filter function that constructs the array with the duplicate values.